app/src/main/java/com/rkm/tasky/repository/abstraction/TaskyReminderRepository.kt

```diff
@@ -0,0 +1,63 @@
+package com.rkm.tasky.repository.abstraction
+
+import com.rkm.tasky.database.model.ReminderEntity
+import com.rkm.tasky.network.model.Reminder
+import com.rkm.tasky.network.util.NetworkError
```

Comment:
I think I mentioned this before 🤔 Your abstraction (domain layer) should not have knowledge about your database or network models (data layer).

You're coupling implementation details with your domain layer which is not allowed. Any DTOs (data transfer objects, like a response model) or DB models must go in your app's data layer since they contain information about the JSON or DB schema. 

The whole goal with domain is that changes in data won't cause changes in domain. However, by putting data models in domain this will be violated, since a change in data (like a change in the JSON schema) would require a change in domain because the fields would need to be renamed.

To fix that, we typically work with mapper functions that take the data model and map it into a domain representation:
```
// In domain
data class LoginUser(
    val accessToken: String,
    val refreshToken: String,
    val fullName: String,
    val userId: String,
    val accessTokenExpirationTimestamp: Long
)

// In data
data class LoginUserResponse(
    val accessToken: String,
    val refreshToken: String,
    val fullName: String,
    val userId: String,
    val accessTokenExpirationTimestamp: Long
)

// In data (LoginMappers.kt)
fun LoginUserResponse.toLoginUser(): LoginUser {
    return LoginUser(...)
}

// In data (AuthRepositoryImpl.kt)
fun login(email: String, password: String): Result<LoginUser> { // <- Returns domain model
    // ...
    return Result.Success(loginUserResponse.toLoginUser()) // <- mapping happens in data layer
}
``` 
For classes like this, this step really feels useless since the domain model looks just like the data model. The important difference is that if there was a change in the JSON schema, just the data model would change, but not the domain model.

In most cases, domain and data models are different though, so don't let this example make you think doing this is always just boilerplate code. Very often, response models work with very simple data which is easy to serialize (imagine a Long timestamp to represent a date - easy to serialize, hard to work with). Mappers can parse such types to data types that are much easier to work with, like `ZonedDateTime`. That makes it very convenient to work with domain models across the app.